sdk/core/azure-core/src/main/java/com/azure/core/implementation/AccessibleByteArrayOutputStream.java

```diff
@@ -30,10 +35,39 @@ public AccessibleByteArrayOutputStream(int initialCapacity) {
     }
 
     @Override
-    public synchronized byte[] toByteArray() {
+    public byte[] toByteArray() {
+        return Arrays.copyOf(buf, count);
+    }
+
+    /**
+     * Returns the internal {@code byte[]} without copying.
+     * <p>
+     * This will be the full {@code byte[]}, so if writing required it to be resized to 8192 bytes but only 6000 bytes
+     * were written the final 2192 bytes will be undefined data. If this is used in an API where a {@code byte[]} is
+     * accepted you must use the range based overload with {@link #count()}, if a range based overload isn't available
+     * use {@link #toByteArray()} which will copy the range of bytes written.
+     *
+     * @return A direct reference to the internal {@code byte[]} where data is being written.
+     */
+    public byte[] toByteArrayUnsafe() {
```

Comment:
Do we need this API? It's best not to expose the underlying raw byte array.

Reply:
This was already exposing the underlying `byte[]` but it was doing it through the overridden `toByteArray()` method, so I added this to make it clear what was being done.

Reply:
Since this is an implementation class, we can just not expose the underlying raw bytes array and if there are any usages in azure-core that require raw byte array, we should refactor to not rely on this behavior.